yt/utilities/lib/bounded_priority_queue.pyx

```diff
@@ -48,13 +68,51 @@ cdef class BoundedPriorityQueue:
 
         return 0
 
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    @cython.initializedcheck(False)
+    cdef int max_heapify_pid(self, np.intp_t index) nogil except -1:
```

Comment:
The implementation of this function looks identical to `max_heapify` above, except it calls `max_heapify_pid` instead of `max_heapify`. Can you explain a little bit why we need all these `_pid` methods on the priority queue and why the implementation isn't different from the regular API?

It might be more maintainable (and more DRY) if we just always did the bookkeeping on IDs or did it optionally (these functions are already kinda branchy).

Reply:
would you prefer `if` statements? The functions are different because I also change the order of the pids as well as the distances - considering the overhead in other places, I doubt if statements would make any noticeable changes to performance. but, I know at the beginning you expressed a desire to keep this function very fast

Reply:
Let's use if statements